File: Data Structures & Algorithms/subsets-ii/submission-0.py
class Solution:
    def subsetsWithDup(self, nums: List[int]) -> List[List[int]]:
        ans=[]
        nums.sort()
        def rec(start,temp):
            ans.append(temp[:])
            for i in range(start,len(nums)):
                if i>start and nums[i]==nums[i-1]:
                    continue
                temp.append(nums[i])
                rec(i+1,temp)
                temp.pop()
         
        rec(0,[])
        return ans
            

File: Data Structures & Algorithms/counting-bits/submission-0.java
class Solution {
    public int[] countBits(int n) {
        int[] ans=new int[n+1];
        for(int i=0;i<n+1;i++){
            int B=i;
            int count=0;
            while(B>0){
                if( (B & 1)>0 ){
                    count++;
                }
                B=B>>1;
            }
            ans[i]=count;
        }
        return ans;
    }
}
